x/auth/account.go

```diff
@@ -115,12 +118,154 @@ func (acc *BaseAccount) SetSequence(seq int64) error {
+	bacc := BaseAccount{
+		Address: addr,
+		Coins:   originalCoins,
+	}
+	return ContinuousVestingAccount{
+		BaseAccount:          bacc,
+		OriginalVestingCoins: originalCoins,
+		StartTime:            startTime,
+		EndTime:              endTime,
+	}
+}
+
+// Implements VestingAccount interface.
+func (vacc ContinuousVestingAccount) IsVesting(blockTime time.Time) bool {
+	return blockTime.Unix() < vacc.EndTime.Unix()
+}
+
+// Implement Vesting Account interface. Uses time in context to calculate how many coins
+// has been released by vesting schedule and then accounts for unlocked coins that have
+// already been transferred or delegated
+func (vacc ContinuousVestingAccount) SendableCoins(blockTime time.Time) sdk.Coins {
+	unlockedCoins := vacc.TransferredCoins
+	scale := sdk.NewDec(blockTime.Unix() - vacc.StartTime.Unix()).Quo(sdk.NewDec(vacc.EndTime.Unix() - vacc.StartTime.Unix()))
+
+	// Add original coins unlocked by vesting schedule
+	for _, c := range vacc.OriginalVestingCoins {
+		amt := sdk.NewDecFromInt(c.Amount).Mul(scale).RoundInt()
+
+		// Must constrain with coins left in account
+		// Since some unlocked coins may have left account due to delegation
+		currentAmount := vacc.GetCoins().AmountOf(c.Denom)
+		if currentAmount.LT(amt) {
+			amt = currentAmount
+			// prevent double count of transferred coins
+			amt = amt.Sub(vacc.TransferredCoins.AmountOf(c.Denom))
+		}
+
+		// Add non-zero coins
+		if !amt.IsZero() {
+			coin := sdk.NewCoin(c.Denom, amt)
+			unlockedCoins = unlockedCoins.Plus(sdk.Coins{coin})
+		}
+	}
+
+	return unlockedCoins
+}
+
+// Implement Vesting Account. Track transfers in and out of account
+// CONTRACT: Send amounts must be negated
+func (vacc *ContinuousVestingAccount) TrackTransfers(coins sdk.Coins) {
+	vacc.TransferredCoins = vacc.TransferredCoins.Plus(coins)
+}
+
+// Implements Vesting Account. Vests all original coins after EndTime but keeps them
+// all locked until that point.
+type DelayTransferAccount struct {
+	BaseAccount
+	TransferredCoins sdk.Coins // Any received coins are sendable immediately
+
+	// All coins unlocked after EndTime
+	EndTime time.Time
+}
+
+func NewDelayTransferAccount(addr sdk.AccAddress, originalCoins sdk.Coins, endTime time.Time) DelayTransferAccount {
+	bacc := BaseAccount{
+		Address: addr,
+		Coins:   originalCoins,
+	}
+	return DelayTransferAccount{
+		BaseAccount: bacc,
+		EndTime:     endTime,
+	}
+}
+
+// Implements VestingAccount
+func (vacc DelayTransferAccount) IsVesting(blockTime time.Time) bool {
+	return blockTime.Unix() < vacc.EndTime.Unix()
+}
+
+// Implements VestingAccount. If Time < EndTime return only net transferred coins
+// Else return all coins in account (like BaseAccount)
+func (vacc DelayTransferAccount) SendableCoins(blockTime time.Time) sdk.Coins {
+	// Check if ctx.Time < EndTime
+	if blockTime.Unix() < vacc.EndTime.Unix() {
```

Comment:
whats the reason for casting these both to unix? They should be comparable in time.Time format. (e.g. `time1.Before(time2)`